bacnet_toolshed/nic_bind.py

```
@@ -142,6 +142,71 @@ def resolve_bacnet_bind(raw: str | None = None, *, default_port: int = _DEFAULT_
+    if len(parts) >= 2:
+        return parts[1].rstrip(":")
+    return ""
+
+
+def list_host_interfaces() -> list[dict[str, str | bool]]:
+    """List IPv4 interfaces for BACnet bind UI (host-side, read-only)."""
+    rows: list[dict[str, str | bool]] = []
+    seen: set[str] = set()
+    try:
+        out = subprocess.check_output(
+            ["ip", "-4", "-o", "addr", "show"],
```

Ruff (0.15.15)

[error] Starting a process with a partial executable path

(S607)